tests/test_component.py

```diff
@@ -60,6 +60,31 @@ def get_template_string(self, context):
                 """
                 return content
 
+            def get_template_data(self, args, kwargs, slots, context):
+                return {
+                    "variable": kwargs.get("variable", None),
+                }
+
+            class Media:
+                css = "style.css"
+                js = "script.js"
+
+        rendered = SimpleComponent.render(kwargs={"variable": "test"})
+        assertHTMLEqual(
+            rendered,
+            """
+            Variable: <strong data-djc-id-ca1bc3e>test</strong>
+            """,
+        )
+
+    # TODO_REMOVE_IN_V2 - `get_context_data()` was superseded by `self.get_template_data`
+    @djc_test(parametrize=PARAMETRIZE_CONTEXT_BEHAVIOR)
+    def test_get_context_data(self, components_settings):
```

Comment:
Added a test specifically for `get_context_data()` to ensure we remain backwards compatible.